src/libraries/System.Private.CoreLib/src/System/ValueTuple.cs

```diff
@@ -504,9 +500,7 @@ public bool Equals(ValueTuple<T1, T2> other)
         /// </remarks>
         bool IStructuralEquatable.Equals(object? other, IEqualityComparer comparer)
         {
-            if (other == null || !(other is ValueTuple<T1, T2>)) return false;
-
-            var objTuple = (ValueTuple<T1, T2>)other;
+            if (other == null || other is not ValueTuple<T1, T2> objTuple) return false;
```

Comment:
a nit: `== null || x is not` patterns can be simplified:
```suggestion
            if (other is not ValueTuple<T1, T2> objTuple) return false;
```

codegen gets slightly improved as well:

<img width="1401" alt="image" src="https://user-images.githubusercontent.com/3840695/99446077-2d92f080-2927-11eb-86d6-f2d19037fa0d.png">

Reply:
Addressed